include/up-cpp/utils/CallbackConnection.h

```diff
@@ -118,79 +150,151 @@ struct [[nodiscard]] Connection {
 	///
 	/// @param args Arguments to be forwarded to the callback
 	template <typename RTT = RT,
-	          typename std::enable_if_t<std::is_same_v<RTT, void>, bool> = true>
-	void operator()(Args... args);
+	          typename std::enable_if_t<std::is_void_v<RTT>, bool> = true>
+	void operator()(Args&&... args) {
+		if (!sever_requested_) {
+			decltype(callback_) callback;
+			{
+				std::lock_guard lk(callback_mtx_);
+				callback = callback_;
+			}
+
+			if (auto locked_cb = callback.lock(); locked_cb) {
+				(*locked_cb)(std::forward<Args>(args)...);
+			}
+
+			notify_if_expired(std::move(callback));
+		}
+	}
```

Comment:
So I just figured out a way to combine both `operator()` versions into a single implementation. Any interest in me adding that to this PR? I think it would probably be better in the long run; it avoids two places where possible fixes could be needed.

Reply:
I suggest putting that change in another PR, this one is close to submission.

Reply:
I'll hold off on this unless there are other changes requested.

Reply:
Change committed. I'll push it once I've addressed more of the comments.